extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -504,24 +504,34 @@ export class AmpConsent extends AMP.BaseElement {
       ) {
         this.updateCacheIfNotNull_(
           response['consentStateValue'],
-          response['consentString'] || undefined
+          response['consentString'] || undefined,
+          response['gdprApplies'] !== undefined
+            ? response['gdprApplies']
+            : !!response['consentRequired']
         );
       }
     });
   }
 
   /**
    * Sync with local storage if consentRequired is true.
+   * Treat all three values as a grouping, only update cache
+   * if responseStateValue is not null
    * @param {string=} responseStateValue
    * @param {string=} responseConsentString
+   * @param {boolean=} responseGdprApplies
    */
-  updateCacheIfNotNull_(responseStateValue, responseConsentString) {
+  updateCacheIfNotNull_(
+    responseStateValue,
+    responseConsentString,
+    responseGdprApplies
+  ) {
     const consentStateValue = convertEnumValueToState(responseStateValue);
-    // consentStateValue and consentString are treated as a pair that will update together
     if (consentStateValue !== null) {
       this.consentStateManager_.updateConsentInstanceState(
         consentStateValue,
-        responseConsentString
+        responseConsentString,
+        responseGdprApplies
```

Comment:
updating gdprApplies here won't work because it requires `consentStateValue` to be not `null`. 
A typical workflow is as following
- consentHref returns `gdprApplies`, but no `consentStateValue`. 
- Then the `consentStateValue` is calculated on the client side. 

I'd recommend we handle the `gdprApplies` separate from consentState and consentString. Like what we do with the dirtyBit. 
Let me know if you have better ideas.

Reply:
Ok, thanks for addressing this. It was either treat it like `consentString`/`consentStateValue` or treat it like `dirtyBit`. I'm good to treat it like `dirtyBit`.